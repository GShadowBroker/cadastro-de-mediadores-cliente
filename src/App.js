import React, { useState, useEffect } from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import Login from "./pages/login/Login";
import Register from "./pages/registration/Register";
import Home from "./pages/Home";
import { login, logout } from "./services/authService";
import { useDispatch, useSelector } from "react-redux";
import { getUser } from "./services/authService";
import { login as runLogin, logout as runLogout } from "./store/authReducer";
import errorHandler from "./utils/errorHandler";

const App = () => {
  const [loading, setLoading] = useState(false);

  const dispatch = useDispatch();
  const session = useSelector((state) => state.authReducer);

  useEffect(() => {
    if (!session.isAuthenticated) {
      setLoading(true);
      getUser()
        .then(({ user }) => {
          dispatch(runLogin(user));
          setLoading(false);
        })
        .catch((err) => {
          setLoading(false);
          console.log(errorHandler(err));
          console.error(err);
        });
    }
  }, [session, dispatch]);

  const handleLogin = async (data) => {
    setLoading(true);
    console.log("credenciais", data);

    try {
      await login(data);
      const { user } = await getUser();
      dispatch(runLogin(user));
      setLoading(false);
    } catch (err) {
      console.error(err);
      setLoading(false);
    }
  };

  const handleLogout = async () => {
    setLoading(true);
    const res = await logout();
    if (res.success) {
      dispatch(runLogout());
      setLoading(false);
    }
  };

  if (session.isAuthenticated) {
    return (
      <Switch>
        <Route path="/protected">
          <div>
            <h1>You are now logged in!</h1>
            {loading ? (
              <button>aguarde...</button>
            ) : (
              <button onClick={handleLogout}>logout</button>
            )}
          </div>
        </Route>
        <Route path="*">
          <Redirect to="/protected" />
        </Route>
      </Switch>
    );
  } else {
    return (
      <Switch>
        <Route path="/login">
          <Login handleLogin={handleLogin} loading={loading} />
        </Route>
        <Route path="/registro">
          <Register />
        </Route>
        <Route>
          <Home path="/" />
        </Route>
        <Route path="*">
          <Redirect to="/login" />
        </Route>
      </Switch>
    );
  }
};

export default App;
