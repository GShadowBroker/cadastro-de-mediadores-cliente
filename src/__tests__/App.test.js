import React from "react";
import { render } from "@testing-library/react";
import { create } from "react-test-renderer";
import App from "../App";
import { unmountComponentAtNode } from "react-dom";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Testing App.js", () => {
  test("renders hello world", () => {
    const { getByText } = render(<App />);
    const linkElement = getByText(/login/i);
    expect(linkElement).toBeInTheDocument();
  });

  test("Taking a snapshot and comparing", () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
