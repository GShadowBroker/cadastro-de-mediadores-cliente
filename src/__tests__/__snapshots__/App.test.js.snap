// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App.js Taking a snapshot and comparing 1`] = `
<h1
  onClick={[Function]}
>
  Hello world!
</h1>
`;
